core/server/services/auth/api-key/admin.js

```diff
@@ -141,7 +141,22 @@ const authenticateWithToken = (req, res, next, {token, JWT_OPTIONS}) => {
             return next(new errors.InternalServerError({err}));
         }
 
-        // authenticated OK, store the api key on the request for later checks and logging
+        // authenticated OK
+
+        if (apiKey.get('user_id')) {
+            // fetch the user and store it on the request for later checks and logging
+            return models.User.findOne({id: apiKey.get('user_id')}).then((user) => {
+                // Personal Tokens shouldn't work for users that aren't active
+                if (user.isActive()) {
+                    req.user = user;
+                }
+                next();
+            }).catch((err) => {
+                next(new errors.InternalServerError({err}));
```

Comment:
Why do we need special handling here as there's same logic at [:163](https://github.com/TryGhost/Ghost/pull/12374/files#diff-bc0bedcac8ec9646d0644c86a91e46f4759bc1b0c2aebac54a2b26ec474c3d15R163) :thinking:

Reply:
Looks like left over from https://github.com/TryGhost/Ghost/pull/12374/commits/4e1d49a9b3ce1e62269a251d4c1e1373361367b8 where we started returning the promise!

Reply:
I've removed the un-needed `catch` :heavy_check_mark: